torchtext/experimental/transforms.py

```diff
@@ -112,3 +112,40 @@ def forward(self, line: str) -> List[str]:
         for regex, replacement_string in self.regex_and_replacement_string_pairs:
             line = regex.Sub(line, replacement_string)
         return line.split()
+
+
+class NgramsTransform(nn.Module):
+    """Return a list of all tokens and ngrams.
+
+    Examples:
+        >>> token_list = ['here', 'we', 'are']
+        >>> ngrams_transform = NgramsTransform()
+        >>> list(ngrams_transform(token_list, 3))
+        >>> ['here', 'we', 'are', 'here we', 'we are', 'here we are']
+
+    """
+    def __init__(self, ):
+        super(NgramsTransform, self).__init__()
+
+    def _get_ngrams(self, token_list: List[str], n: int) -> List[List[str]]:
+        ngrams: List[List[str]] = [token_list[i:i + n] for i in range(len(token_list) - n + 1)]
+        return ngrams
+
+    def forward(self, token_list: List[str], ngrams: int) -> List[str]:
```

Comment:
`ngrams` should be defined in constructor. For a ngrams transform instance, it should work with a constant `ngrams`.

Reply:
Done